polygon/sync/difficulty.go

```diff
@@ -0,0 +1,54 @@
+package sync
+
+import (
+	lru "github.com/hashicorp/golang-lru/arc/v2"
+	"github.com/ledgerwatch/log/v3"
+
+	"github.com/ledgerwatch/erigon-lib/chain"
+	libcommon "github.com/ledgerwatch/erigon-lib/common"
+	"github.com/ledgerwatch/erigon/consensus/bor"
+	heimdallspan "github.com/ledgerwatch/erigon/consensus/bor/heimdall/span"
+	"github.com/ledgerwatch/erigon/consensus/bor/valset"
+	"github.com/ledgerwatch/erigon/core/types"
+)
+
+type DifficultyCalculator interface {
+	HeaderDifficulty(header *types.Header) (uint64, error)
+}
+
+type difficultyCalculatorImpl struct {
+	borConfig       *chain.BorConfig
+	span            *heimdallspan.HeimdallSpan
+	signaturesCache *lru.ARCCache[libcommon.Hash, libcommon.Address]
+}
+
+func NewDifficultyCalculator(
+	borConfig *chain.BorConfig,
+	span *heimdallspan.HeimdallSpan,
+) DifficultyCalculator {
+	signaturesCache, err := lru.NewARC[libcommon.Hash, libcommon.Address](4096)
```

Comment:
this is a question just out of curiosity - what is the thought process is behind picking the size to be 4096 for this ecrecover cache?

Reply:
I don't know. I copied the value of `inmemorySignatures` constant used for the same purpose. I'll make the constant public to reuse it here.